test/bin/run-tests-ubuntu.sh

```diff
@@ -1,76 +1,165 @@
+        # TODO Enable Refactor and Convention reports
+        # TODO Move --disable into rc
+        pylint --reports=no --disable=C,R,W0312,W0141,W0110 api
+
+        # echo "Running pep8 ..."
+        # pep8 --max-line-length=150 --ignore=E402 api
+    fi
+
+    if ${RUN_UNIT}; then
+        echo "Running unit tests ..."
+        rm -f .coverage
+        py.test --cov=api --cov-report= test/unit_tests/python $PYTEST_ARGS
+    fi
+
+    if ${RUN_INTEG}; then
+        echo "Running integration tests ..."
+        uwsgi --http "localhost:8081" --master --http-keepalive \
+            --so-keepalive --add-header "Connection: Keep-Alive" \
+            --processes 1 --threads 1 \
+            --enable-threads \
+            --wsgi-file bin/api.wsgi \
+            --die-on-term \
+            --logformat '%(addr) - %(user) [%(ltime)] "%(method) %(uri) %(proto)" %(status) %(size) "%(referer)" "%(uagent)" request_id=%(request_id)' \
+            --env "SCITRAN_PERSISTENT_DB_URI=$SCITRAN_PERSISTENT_DB_URI" \
+            --env "SCITRAN_PERSISTENT_DB_LOG_URI=$SCITRAN_PERSISTENT_DB_LOG_URI" \
+            --env "SCITRAN_PERSISTENT_PATH=$SCITRAN_PERSISTENT_PATH" \
+            --env "SCITRAN_PERSISTENT_DATA_PATH=$SCITRAN_PERSISTENT_DATA_PATH" \
+            --env "SCITRAN_CORE_DRONE_SECRET=$SCITRAN_CORE_DRONE_SECRET" \
+            --env "SCITRAN_RUNTIME_COVERAGE=true" \
+            --env "SCITRAN_CORE_ACCESS_LOG_ENABLED=true" &
+        export API_PID=$!
+
+        echo "Connecting to API"
+        until $(curl --output /dev/null --silent --head --fail "$SCITRAN_SITE_API_URL"); do
+            printf '.'
+            sleep 1
+        done
+
+        py.test test/integration_tests/python $PYTEST_ARGS
+
+
+        # Create resources that Abao relies on
+        python test/integration_tests/abao/load_fixture.py
+
+        # If no VIRTUAL_ENV, make sure /usr/local/bin is in the path
+        if [[ -z "${VIRTUAL_ENV:-}" ]]; then
+            PATH="/usr/local/bin:$PATH"
+            npm install test/integration_tests
+        else
+            npm install --global test/integration_tests
+        fi
+
+        PATH="$(npm bin):$PATH"
+
+        # Allow us to require modules from package.json,
+        # since abao_test_hooks.js is not being called from the package directory
+        integration_test_node_modules="$(pwd)/node_modules/scitran-core-integration-tests/node_modules"
+
+        # Have to change into definitions directory to resolve
+        # relative $ref's in the jsonschema's
+        pushd raml/schemas/definitions
+        NODE_PATH="$integration_test_node_modules" abao ../../api.raml "--server=$SCITRAN_SITE_API_URL" "--hookfiles=../../../test/integration_tests/abao/abao_test_hooks.js"
+        popd
+    fi
+}
+
+
+function clean_up () {
+    export TEST_RESULT_CODE=$?
+    set +e
+
+    echo
+    echo "Test return code = $TEST_RESULT_CODE"
+
+    if [[ -n "${API_PID:-}" ]]; then
+        # Killing uwsgi
+        kill $API_PID || true
```

Comment:
What's the purpose of `|| true`? It would be obvious to me if exit on error was still set.